js/router.js

```diff
@@ -1,15 +1,147 @@
+    super(options);
+    this.options = options;
+
+    // Temporarily requiring a dummy 'users' collection. Eventually, we'll
+    // be getting this data via fetches to the server
+    if (!options.usersCl) {
+      throw new Error('Need me some dummy users!');
+    }
+
+    this.usersCl = options.usersCl;
+
+    if (!options.pageNavVw) {
+      throw new Error('Please provide the app\'s pageNavVw.');
+    }
+
+    this.pageNavVw = options.pageNavVw;
+
     const routes = [
-      // ['home/:state(/:searchText)', 'home'],
-      // ['userPage/:userID(/:state)(/:itemHash)(/:skipNSFWmodal)', 'userPage'],
-      // [/^@([^\/]+)(.*)$/, 'userPageViaHandle'],
-      // ['transactions/:state(/:orderID)(/:tabState)', 'transactions'],
-      // ['settings/:state', 'settings'],
+      [/^@([^\/]+)[\/]?([^\/]*)[\/]?([^\/]*)[\/]?([^\/]*)$/, 'userViaHandle'],
+      [/^(Qm[a-zA-Z0-9]+)[\/]?([^\/]*)[\/]?([^\/]*)[\/]?([^\/]*)$/, 'user'],
+      ['transactions', 'transactions'],
+      ['transactions/:tab', 'transactions'],
+      ['*path', 'pageNotFound'],
     ];
 
-    routes.forEach((route) => this.route.apply(this, route));
+    routes.slice(0)
+      .reverse()
+      .forEach((route) => this.route.apply(this, route));
+
+    this.setAddressBarText();
+
+    $(window).on('hashchange', () => {
+      this.setAddressBarText();
+    });
+  }
+
+  setAddressBarText() {
+    if (location.hash.startsWith('#transactions')) {
+      this.pageNavVw.setAddressBar();
+    } else {
+      this.pageNavVw.setAddressBar(location.hash.slice(1));
+    }
+  }
+
+  execute(callback, args) {
+    getPageContainer().empty();
+
+    if (callback) callback.apply(this, args);
+  }
+
+  loadPage(vw) {
+    if (this.currentPage) {
+      this.currentPage.remove();
+    }
+
+    this.currentPage = vw;
+    getPageContainer().append(vw.el);
+  }
+
+  // Temporary fudge, since we're actually hitting the one name api,
+  // but I'd like the guids to match the 2.0 format of being prefaced with Qm.
+  // Once we're getting the data from the server, this method will go bye-bye
+  // and will directly call getGuid from the util module.
+  getGuid(guid) {
+    const deferred = $.Deferred();
+
+    getGuid(guid).done((id) => deferred.resolve(`Qm${id}`))
+      .fail((...args) => deferred.reject(...args));
+
+    return deferred.promise();
+  }
+
+  getUser(guid) {
+    const deferred = $.Deferred();
+    const user = this.usersCl.get(guid);
+
+    // setting timeout to simulate the latency of an async call
+    setTimeout(() => {
+      if (user) {
+        deferred.resolve(user);
+      } else {
+        deferred.reject();
+      }
+    }, parseInt(Math.random() * 1000, 10));
+
+    return deferred.promise();
+  }
+
+  userViaHandle(handle, ...args) {
+    this.getGuid(handle).done((guid) => {
+      this.user(guid, ...args);
+    }).fail(() => {
+      this.userNotFound();
+    });
+  }
+
+  user(guid, tab, ...args) {
+    tab = tab || 'store'; // eslint-disable-line no-param-reassign
```

Comment:
The linter is not happy because we should really be using ES6's default arg syntax here where `tab = 'store'` could be declared right in the function definition. The problem is that only assigns the default if the argument comes in as `undefined`, whereas the Router is sending unmatched path fragments as `null`. So, we're putting in the lint override and doing it the old school way.